internal/runtime/types.go

```diff
@@ -0,0 +1,30 @@
+/**
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+**/
+
+package runtime
+
+type Runtime interface {
```

Comment:
How is this different from the interface and implementations here: https://github.com/NVIDIA/nvidia-container-toolkit/blob/bead6f98f3e99cc7e2b9140d4e860496b1690a79/pkg/config/engine/api.go#L20

Reply:
Yes it looks like we can reuse it.

Reply:
I might be able to reuse that implementation by tweaking it a bit.

Reply:
@elezar In my opinion, updating the container toolkit implementations to add Kata-related runtime options will unnecessarily complicate the code without providing any significant advantage.

Reply:
I wasn't suggesting that you use that implementation. I just wanted to know what the differences were. 

We initially made the config package public in the toolkit to allow for reuse with the kata manager. I was under the impression that @ArangoGutierrez and @cdesiniotis who were actively working on it would contribute functionality there as required. Since this has not happened I understand that this may not serve your needs at present.

Reply:
Yes, initially we were planning to reuse the public config package from the toolkit. But there were enough differences between what was required for kata vs container toolkit (at least for containerd) that we decided to maintain a separate implementation of this package in the kata-manager repository.